opm/simulators/timestepping/AdaptiveTimeStepping_impl.hpp

```diff
@@ -225,7 +225,7 @@ namespace Opm {
             if( timestep_verbose_ )
             {
                 std::ostringstream ss;
-                ss <<"  Substep " << substepTimer.currentStepNum() << ", stepsize "
```

Comment:
I think this will be misleading, as we are here writing information of a substep of a report step. The number that is written out here starts from 0 and counts up just for the current report step. I think it is confusing to read "Time step 0" at 1000 days without any indication that it is a substep. What about using the term sub-timestep instead (or just keeping substep)?

Also, the spaces at the start of the string are intended to make the major info stand out more, substep info is indented compared to report step info. I am not saying it is perfect, just asking: did you intend to change that as well?

This also applies to line 312 below.

Reply:
We may misunderstand each other. Report step simply means that the fluid in place report is written. It is not a time step. The time steps are all time steps, and all time steps are here reported as time steps I believe. Maybe it is the counter that is confusing, since it counts timesteps since the last time step that was also a report step? We can simply remove the counter, or maybe start counting from 1? or replace the counter with a total time since start?

Reply:
I think that time step is often used synonymously with report step, contributing to confusion here. For example, when viewing result data in ResInsight the step numbers correspond to report steps, not actual numerical (sub)steps taken. Therefore we should keep outputting clearly and with focus those step numbers (but calling it Report step as you propose here).

We could also maintain a total (including substeps) step count and output that, however I think there is a risk for confusion there.

Changing the substep counter output to start from 1 instead of 0 is ok with me, but removing the counter makes it hard to understand what is happening (especially if a step is chopped) so I am against that. I think *adding* the current time since start to the substep output is a good idea, I support that.

Reply:
> I think that time step is often used synonymously with report step

Yes, but even that is actually wrong and confusing for anybody but flow developers. Report step here only means that fluid in place numbers are written, that is the report. Maybe we should make the message more explicit? I am currently trying to understand the output writer code to get a separate message in place clearly stating when restart data is written too. Check out model 2, restart data steps are different from report steps. For 3D you will only see restart steps in resinsight, but for 2D plotting all time steps are written. I am more than happy to make the messages more verbose to avoid confusion, but I fear that much of this confusion is created by ourselves by not being explicit from day one. This confusion has been bugging me for quite some time, so I hope we can clear it all up by release.

Sure, we can keep the counter, and I can add the current time to it. There is after all plenty space left on that line for additional verbosity.